tests/configurable_cluster.py

```diff
@@ -193,30 +207,51 @@ def copy_to_host(self, source_path, host, dest_path=None):
         if not dest_path:
             dest_path = os.path.join(self.mount_dir,
                                      os.path.basename(source_path))
-        self.exec_cmd_on_host(host, 'mkdir -p ' + os.path.dirname(dest_path))
+        self.exec_cmd_on_host(host, 'mkdir -p {dir}'.format(dir=os.path.dirname(dest_path)))
+        self.exec_cmd_on_host(host, 'chown {user}:{group} {file}'.format(
+            user=self.user, group=self.user, file=os.path.dirname(dest_path)))
 
         ssh = paramiko.SSHClient()
         ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
-        ssh.connect(host, username=self.user, password=self.password,
+        ssh.connect(host, username=self.user, key_filename=self.key_path,
                     timeout=180)
+
+        # Upload to dummy location because paramiko doesn't allow SFTP using
+        # sudo when logged in as a non root user. Due to this limitation, Fabric
+        # uses the same methodology to upload files.
+        dummy_path = '/tmp/{random_dir}/{dest_dir}'.format(
+            random_dir=str(uuid.uuid1()), dest_path=os.path.basename(dest_path))
+        self.exec_cmd_on_host(host, 'mkdir -p {dir}'.format(dir=os.path.dirname(dummy_path)))
+        self.exec_cmd_on_host(host, 'chown {user}:{group} {file}'.format(
+            user=self.user, group=self.user, file=os.path.dirname(dummy_path)))
         sftp = ssh.open_sftp()
-        sftp.put(source_path, dest_path)
+        sftp.put(source_path, dummy_path)
         sftp.close()
+
+        # Move to final location using sudo
+        self.exec_cmd_on_host(host, 'mv {source} {dest}'.format(source=dummy_path, dest=dest_path))
+
+        # Remove dummy path directory
+        self.exec_cmd_on_host(host, 'rm -rf {dir}'.format(dir=os.path.dirname(dummy_path)))
```

Comment:
This seems like a good candidate for a context manager, but I'm fine with it like this for test code.